src/viam/trajex/totg/trajectory.cpp

```diff
@@ -439,40 +440,123 @@ enum class integration_event : std::uint8_t {
+    while (boundary_cursor.position() < path_length) {
+        auto current_segment = *boundary_cursor;
+        const arc_length boundary = current_segment.end();
+
+        // Make sure we are not at the end of the path
+        if (boundary >= path_length) {
+            break;
+        }
+
+        // Only check boundaries ahead of our starting position
+        if (boundary <= cursor.position()) {
+            boundary_cursor.seek(boundary);
+            continue;
+        }
+
+        // Sample geometry from segment ending at boundary
+        const auto q_prime_before = current_segment.tangent(boundary);
+        const auto q_double_prime_before = current_segment.curvature(boundary);
+
+        // Advance cursor to boundary (moves to next segment)
+        boundary_cursor.seek(boundary);
+        auto segment_after = *boundary_cursor;
+
+        // Sample geometry from segment starting at boundary
+        const auto q_prime_after = segment_after.tangent(boundary);
+        const auto q_double_prime_after = segment_after.curvature(boundary);
+
+        // Compute velocity limits on both sides
+        const auto [s_dot_max_acc_before, s_dot_max_vel_before] =
+            compute_velocity_limits(q_prime_before, q_double_prime_before, opt.max_velocity, opt.max_acceleration, opt.epsilon);
+
+        const auto [s_dot_max_acc_after, s_dot_max_vel_after] =
+            compute_velocity_limits(q_prime_after, q_double_prime_after, opt.max_velocity, opt.max_acceleration, opt.epsilon);
+
+        // Skip if velocity limits are degenerate on either side
+        if (std::abs(s_dot_max_vel_before) < opt.epsilon || std::abs(s_dot_max_vel_after) < opt.epsilon) {
+            continue;
+        }
+
+        // Compute curve slopes on both sides (equation 37)
+        const double curve_slope_before =
+            compute_velocity_limit_derivative(q_prime_before, q_double_prime_before, opt.max_velocity, opt.epsilon);
+
+        const double curve_slope_after =
+            compute_velocity_limit_derivative(q_prime_after, q_double_prime_after, opt.max_velocity, opt.epsilon);
+
+        // Compute minimum accelerations at velocity limits on both sides
+        const auto [s_ddot_min_before, s_ddot_max_before] =
+            compute_acceleration_bounds(q_prime_before, q_double_prime_before, s_dot_max_vel_before, opt.max_acceleration, opt.epsilon);
+
+        const auto [s_ddot_min_after, s_ddot_max_after] =
+            compute_acceleration_bounds(q_prime_after, q_double_prime_after, s_dot_max_vel_after, opt.max_acceleration, opt.epsilon);
+
+        // Check equations 41 and 42:
+        // (s_ddot_min(s-, s_dot_max_vel(s-)) / s_dot >= d/ds s_dot_max_vel(s-))  (41)
+        // AND
+        // (s_ddot_min(s+, s_dot_max_vel(s+)) / s_dot <= d/ds s_dot_max_vel(s+))  (42)
+        const double trajectory_slope_before = s_ddot_min_before / s_dot_max_vel_before;
+        const double trajectory_slope_after = s_ddot_min_after / s_dot_max_vel_after;
```

Comment:
I don't see an `s_dot` in the denominator in eq 41 and 42 in the paper (there is a similar expression VI.3 which does have an s_dot in the denominator, but that's about following the limit curve). Or do you think this is another typo in the paper?

Or, if this is correct, is VI.3 wrong somehow? What does dimensional analysis say?

(I find myself wanting the strong types more and more)